trollflow2/plugins/__init__.py

```diff
@@ -296,16 +323,106 @@ def save_datasets(job):
     base_config = job['input_mda'].copy()
     base_config.pop('dataset', None)
     eager_writing = job['product_list']['product_list'].get("eager_writing", False)
-    with renamed_files() as renames:
+    early_moving = job['product_list']['product_list'].get("early_moving", False)
+    sentinel = object()
+    call_on_done = job["product_list"]["product_list"].get("call_on_done", sentinel)
+    if call_on_done is not sentinel:
+        callbacks = [dask.delayed(c) for c in call_on_done]
+    else:
+        callbacks = None
+    with renamed_files(not early_moving) as renames:
         for fmat, fmat_config in plist_iter(job['product_list']['product_list'], base_config):
-            obj = save_dataset(scns, fmat, fmat_config, renames, compute=eager_writing)
-            if obj is not None:
-                objs.append(obj)
+            late_saver = save_dataset(scns, fmat, fmat_config, renames, compute=eager_writing)
+            late_saver = _maybe_apply_callbacks(late_saver, callbacks, job, fmat_config)
```

Comment:
Sounds like it's random whether the callbacks are called or not :sweat_smile: I think we've named this kind of conditionals simply `_apply_foo()` or such in other packages.

Reply:
Renamed to `_apply_callbacks` in https://github.com/pytroll/trollflow2/pull/168/commits/1f56dd1f6f279aec198118f7cb3d4f0a02aadebf